src/main/resources/rules/AEM-18.md

```diff
@@ -0,0 +1,27 @@
+The [`page.getContentResource()`](https://helpx.adobe.com/experience-manager/6-3/sites/developing/using/reference-materials/javadoc/com/day/cq/wcm/api/Page.html#getContentResource())
+is a method of the [`Page`](https://helpx.adobe.com/experience-manager/6-3/sites/developing/using/reference-materials/diff-previous/changes/com.day.cq.wcm.api.Page.html)
+which allows you to get a Resource from the page.
+It is possible to get a `null` if jcr:content does not exist.
+
+Lack of jrc:content can be caused by not publishing the parent page.
+
+You should always null check the return value of getContentResource()
+
+== Noncompliant Code Example
+
+```
+public ValueMap getProperty(Page page) {
+	return page.getContentResource().getValueMap();
+}
+```
+
+== Compliant Solution
+
+```
+public ValueMap getProperty(Page page) {
+	if (page.getContentResource != null) {
+	  return page.getContentResource.getValueMap();
```

Comment:
Another compliant solution could use an `Optional`. While I find it an overkill for such a simple case, it's still a valid solution. Will it be detected by this rule as a compliant one?

Reply:
Ah, yes :)
Still, it may be worth documenting here in the README

Reply:
== Another Compliant Solution
```
private ValueMap getProperty(Page page) {
    Optional<Resource> contentResourceOptional = Optional.ofNullable(page.getContentResource());
    return contentResourceOptional.get().getValueMap();
}
```
Is it fine?